### ISSUE TITLE
Frequent crashes with "Wrong parameters for MeiliSearch\Exceptions\ApiException"

### ISSUE THREAD

**Issue by Tarpsvo** [meilisearch/meilisearch-php#171]
As ironic as it is, it seems ApiException creation fails due to invalid parameters and crashes.

Here's an image of the stack trace from Sentry:
![image](https://user-images.githubusercontent.com/2018660/119138052-253c7f00-ba4a-11eb-9643-b6468c6eb73f.png)

Seems like Client.php uses invalid order for the params in parseResponse error handler.

---- Update:
Nevermind, it seems like the order of parameters is correct for the ApiException constructor. I'll try to investigate further.

On a sidenote: we're using PHP 7.4.

**Comment by Tarpsvo** [comment]
From what I see, PHP seems to call the ApiException constructor in Exception class as well? I guess it doesn't like the parent::__constructor for some reason?

**Comment by curquiza** [comment]
Hello! Thanks for the report. What is the version of MeiliSearch?
What is the version of meilisearch-php?
Can you send us the steps you made to reproduce the issue?

**Comment by Tarpsvo** [comment]
Hi!

We're using meilisearch-php version 0.18.0.

Unfortunately, I do not have any steps as all the errors are reported to Sentry error logging from our background job runners in development and production environments. I'll try to look into this issue and debug it a bit more.

You can see the full stacktrace here:
https://sentry.io/share/issue/67a074503d7f453a933c05560b73c5a4/

**Comment by Tarpsvo** [comment]
Seems like the shortest way to reproduce this is by using tinker:
![image](https://user-images.githubusercontent.com/2018660/119145493-5de05680-ba52-11eb-99fa-5c7a546b45dd.png)

**Comment by bidoubiwa** [comment]
The error seemed to be thrown when laravel scout updates and send the update to MeiliSearch's `addDocuments` method. 
<img width="947" alt="Screenshot 2021-05-21 at 20 24 57" src="https://user-images.githubusercontent.com/33010418/119182091-c808f380-ba72-11eb-8d20-68461847d161.png">

This is the code on the laravel scout: 
```php
 public function update($models)
    {
        if ($models->isEmpty()) {
            return;
        }

        $index = $this->meilisearch->index($models->first()->searchableAs());

        if ($this->usesSoftDelete($models->first()) && $this->softDelete) {
            $models->each->pushSoftDeleteMetadata();
        }

        $objects = $models->map(function ($model) {
            if (empty($searchableData = $model->toSearchableArray())) {
                return;
            }

            return array_merge($searchableData, $model->scoutMetadata());
        })->filter()->values()->all();

        if (! empty($objects)) {
            $index->addDocuments($objects, $models->first()->getKeyName());
        }
    }
```

And this is the code MeiliSearch side: 

```php
public function addDocuments(array $documents, ?string $primaryKey = null)
    {
        return $this->http->post(self::PATH.'/'.$this->uid.'/documents', $documents, ['primaryKey' => $primaryKey]);
    }
```

I supposed the error might come from the `primaryKey` used as optional parameter, but it wasn't. 

:( Tell us if you find something

**Comment by Tarpsvo** [comment]
Did you try creating a new ApiException using `php artisan tinker`? It seems to me that the constructor is somehow scuffed, but I can't understand why.

**Comment by Tarpsvo** [comment]
Ah okay - so I see what's crashing here. It seems that parent::__construct() doesn't want to receive null as the first argument. This simple fallback prevents that issue:

```
public function __construct($httpStatus, $httpBody, $previous = null)
    {
        $this->httpBody = $httpBody;
        $this->httpStatus = $httpStatus;
        $this->message = $this->getMessageFromHttpBody();
        $this->errorCode = $this->getErrorCodeFromHttpBody();
        $this->errorLink = $this->getErrorLinkFromHttpBody();
        $this->errorType = $this->getErrorTypeFromHttpBody();

        parent::__construct($this->message ?? '', $this->httpStatus, $previous);
    }
```

I'll also try to look into what the Meilisearch API is returning in this case.

**Comment by bidoubiwa** [comment]
Do you know in what circumstances the message was `null`? As it might be a bug from meilisearch-core

**Comment by curquiza** [comment]
I don't think it comes from the core. It might happen if the server (for example NGINX) wraps the meilisearch server -> it will only send an HTTP status but no body with `message` inside.

We should indeed return a message by default in the `getMessageFromHttpBody` method:
https://github.com/meilisearch/meilisearch-php/blob/f8b9831d635443b5fd28b83b4675c43f9b4b3c2c/src/Exceptions/ApiException.php#L53-L60

I think the best would be to catch the "reason phrase", I mean use the `getReasonPhrase()` method ([see this docs of `Psr\Http\Message\ResponseInterface`](https://www.php-fig.org/psr/psr-7/)) as we already get the status with `getStatusCode` method in this part of the code: 
https://github.com/meilisearch/meilisearch-php/blob/f8b9831d635443b5fd28b83b4675c43f9b4b3c2c/src/Http/Client.php#L186-L193

**Comment by Tarpsvo** [comment]
Hi @curquiza!

Thank you for your reply. I opened a PR and followed your suggestions when creating the fix for this issue. I also added a new test that covers this and validates that it doesn't happen again.

If the fallback here (the selected text) were to be removed, the new test will fail (just to prove the issue was there).
![image](https://user-images.githubusercontent.com/2018660/119652535-e5e9a600-be2e-11eb-81b2-e5d48ed0a5ef.png)

The failed test does produce the error I encountered.
```
1) Tests\Exceptions\ApiExceptionTest::testExceptionWithNoHttpBody
Error: Wrong parameters for MeiliSearch\Exceptions\ApiException([string $message [, long $code [, Throwable $previous = NULL]]])
```
